### PR TITLE
remove mingw32.mingw.cmake

### PR DESCRIPTION
**Pull request by lasconic**
Counterpart of PR #3213 for master

### CODE REVIEW

**Comment by AntonioBL** [comment]
I think that Qt5WebChannel (and its dll) is no more used, see https://github.com/musescore/MuseScore/blob/3798cffcc6b917ea08e1001618aa8dad82cb4cfa/build/FindQt5.cmake#L1

**Comment by Jojo-Schmitz** [comment]
The removal of Qt5CLucene.dll would mean we need Qt-5.9 or later. Shouldn't that get done in https://github.com/musescore/MuseScore/blob/master/CMakeLists.txt#L51, or similar to that in a mingw section?

**Comment by lasconic** [comment]
*The removal of Qt5CLucene.dll would mean we need Qt-5.9 or later.*
On Windows only. Ideally, I would like this dll list to be dependent on the list in `findQt5`. So it's still need work. Also, there should be a way to link the release dll even in debug more to avoid the lag mentioned by @AntonioBL... So consider this PR an experiment for now.

**Comment by Jojo-Schmitz** [comment]
Qt 5.9.2 is expected any day now (well, next week actually)

**Comment by Jojo-Schmitz** [comment]
Qt 5.9.2 just got released

**Comment by lasconic** [comment]
Now the dlls are not listed anymore but provided by CMake. Also we link and deploy release version of the Qt libraries. It looks good to me and I really like to be able to choose the Qt version via `%PATH%` only.

**Comment by lasconic** [comment]
Appveyor is failing because of https://help.appveyor.com/discussions/problems/10253-maximum-allowed-artifact-storage-size-of-1000-mb-will-be-exceeded I hope they will increase the storage size.

**Comment by Jojo-Schmitz** [comment]
? To me AppVeyor looks like having succeeded

File: CMakeLists.txt [hunk 1]
```diff
@@ -169,8 +169,8 @@ if (APPLE)
 else (APPLE)
    if (MINGW)
       # -mno-ms-bitfields see #22048
-      set(CMAKE_CXX_FLAGS_DEBUG   "-std=c++11 -mno-ms-bitfields -g")
-      set(CMAKE_CXX_FLAGS_RELEASE "-std=c++11 -mno-ms-bitfields -O2 -DNDEBUG -DQT_NO_DEBUG")
+      set(CMAKE_CXX_FLAGS_DEBUG   "-std=gnu++11 -mno-ms-bitfields -g")
+      set(CMAKE_CXX_FLAGS_RELEASE "-std=gnu++11 -mno-ms-bitfields -O2 -DNDEBUG -DQT_NO_DEBUG")
```
Comment (Jojo-Schmitz) [inline CMakeLists.txt]: This change seems to result in several new warnings. Not necessarily a bad thing ;-) but something should be done about those

Comment (lasconic) [reply CMakeLists.txt]: Without this change, the precompiled header doesn't work with Qt 5.10. Cmake automatically adds this flag when using Qt but the flag is not added to the compilation of the precompiled header. If the flags are different the PCH is  ignored.

**Comment by lasconic** [comment]
Yes, they fixed it just before I went to bed, so I restarted the build.

File: CMakeLists.txt (hunk 1 above)
Comment (Jojo-Schmitz) [reply CMakeLists.txt]: Hmm, I don't see these warning when using that change with Qt 5.8.x here.

Comment (Jojo-Schmitz) [reply CMakeLists.txt]: Yes, just those. And only on AppVeyour, locally it is just 2 warnings one unused function and one unused parameter

Comment (lasconic) [reply CMakeLists.txt]: Ok nothing severe.

**Comment by lasconic** [comment]
OK. Let's try it on master. I don't rebase on purpose.

**Pull request merged**